### PR TITLE
reader: init at 0.4.5

### LINKED ISSUE #344420: Package request: reader
**Project description**

reader is for your command line what the “readability” view is for modern browsers: A lightweight tool offering better readability of web pages on the CLI.

**Metadata**

* homepage URL: https://github.com/mrusme/reader
* source URL: https://github.com/mrusme/reader
* license: gpl3
* platforms: linux, darwin(?)

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by theobori**
## Description of changes

Fixes #344420

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/re/reader/package.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      owner = "mrusme";
+      repo = finalAttrs.pname;
+      rev = "v${finalAttrs.version}";
+      hash = "sha256-9hZ7ZS+p6PoLKcuHS2re537wxojN2SzhOm5gBuRX9Xc=";
+    };
+
+    vendorHash = "sha256-obYdifg3WrTyxgN/VtzgpL31ZOyPNtVT8UDQts0WodQ=";
+
+    meta = {
+      description = "reader is for your command line what the “readability” view is for modern browsers: A lightweight tool offering better readability of web pages on the CLI";
```
Comment (AndersonTorres) [inline pkgs/by-name/re/reader/package.nix:21]:
```suggestion
      description = "Lightweight tool offering better readability of web pages on the CLI";
```

File: pkgs/by-name/re/reader/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      homepage = "https://github.com/mrusme/reader";
+      changelog = "https://github.com/mrusme/reader/releases";
+      license = lib.licenses.gpl3;
```
Comment (AndersonTorres) [inline pkgs/by-name/re/reader/package.nix:24]:
```suggestion
      license = lib.licenses.gpl3Plus;
```

File: pkgs/by-name/re/reader/package.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  buildGoModule,
+}:
+let
+  finalAttrs = {
+    pname = "reader";
+    version = "0.4.5";
+
+    src = fetchFromGitHub {
+      owner = "mrusme";
+      repo = finalAttrs.pname;
```
Comment (AndersonTorres) [inline pkgs/by-name/re/reader/package.nix:13]:
```suggestion
      repo = "reader";
```

File: pkgs/by-name/re/reader/package.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildGoModule,
+}:
+let
+  finalAttrs = {
```
Comment (AndersonTorres) [inline pkgs/by-name/re/reader/package.nix:7]: A better structure:

```suggestion
  self = buildGoModule {
```

File: pkgs/by-name/re/reader/package.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    pname = "reader";
+    version = "0.4.5";
+
+    src = fetchFromGitHub {
+      owner = "mrusme";
+      repo = finalAttrs.pname;
+      rev = "v${finalAttrs.version}";
```
Comment (AndersonTorres) [inline pkgs/by-name/re/reader/package.nix:14]:
```suggestion
      rev = "v${self.version}";
```

File: pkgs/by-name/re/reader/package.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      description = "reader is for your command line what the “readability” view is for modern browsers: A lightweight tool offering better readability of web pages on the CLI";
+      homepage = "https://github.com/mrusme/reader";
+      changelog = "https://github.com/mrusme/reader/releases";
+      license = lib.licenses.gpl3;
+      maintainers = with lib.maintainers; [ theobori ];
+      mainProgram = "reader";
+    };
+  };
+in
+buildGoModule finalAttrs
```
Comment (AndersonTorres) [inline pkgs/by-name/re/reader/package.nix:30]:
```suggestion
self
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]
LGTM

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**